envi/archs/rx/disasm.py

```diff
@@ -0,0 +1,1349 @@
+import sys
+import copy
+import struct
+import logging
+
+import envi
+import envi.exc as e_exc
+import envi.bits as e_bits
+import envi.memory as e_mem
+
+from . import regs, rxtables
+from .const import *
+
+logger = logging.getLogger(__name__)
+
+
+class RxOpcode(envi.Opcode):
+
+    def __init__(self, va, opcode, mnem, operands, iflags=0, size=0):
+        self.iflags = iflags | envi.ARCH_RX
+        envi.Opcode.__init__(self, va, opcode, mnem, 0, size, operands, iflags)
+
+    def __len__(self):
+        return self.size
+
+
+    def __repr__(self):
+        """
+        Render this opcode to the specified memory canvas
+        """
+        mnem = self.mnem
+        if self.iflags & IF_BYTE:
+            mnem += '.b'
+        elif self.iflags & IF_WORD:
+            mnem += '.w'
+        elif self.iflags & IF_LONG:
+            mnem += '.l'
+        elif self.iflags & IF_UWORD:
+            mnem += '.uw'
+        elif self.iflags & IF_24BIT:
+            mnem += '.a'
+        elif self.iflags & IF_SMALL:
+            mnem += '.s'
+
+
+        return mnem + " " + ",".join([o.repr(self) for o in self.opers])
+
+    def render(self, mcanv):
+        """
+        Render this opcode to the specified memory canvas
+        """
+        mnem = self.mnem
+        if self.iflags & IF_BYTE:
+            mnem += '.b'
+        elif self.iflags & IF_WORD:
+            mnem += '.w'
+        elif self.iflags & IF_LONG:
+            mnem += '.l'
+        elif self.iflags & IF_UWORD:
+            mnem += '.uw'
+
+        mcanv.addNameText(mnem, typename="mnemonic")
+        mcanv.addText(" ")
+
+        # Allow each of our operands to render
+        imax = len(self.opers)
+        lasti = imax - 1
+        for i in range(imax):
```

Comment:
Why not `for idx, oper in enumerate(image)`?

Reply:
great idea!